modules/ipc/ipcProviderBackend.js

```diff
@@ -1,670 +1,400 @@
+                        }, {
+                            timeout: 5000
+                        })
+                        .then(() => {
+                            log.debug(`Tell owner ${item.id} that socket is again writeable`);
+
+                            item.owner.send('ipcProvider-setWritable', true);
+                        });
+                    }
+                }))
+                .catch((err) => {
+                    log.error('Error re-connecting sockets', err);
+                });
 
-        } else {
-            this.sender.send('ipcProvider-setWritable', true);
+                break;
         }
-
-    };
+    }
 
     /**
-    Creates the socket and sets up the listeners.
-
-    @method setupSocket
-    */
-    GethConnection.prototype.setupSocket = function() {
-        var _this = this;
-        
+     * Handle IPC call to send a request.
+     * @param  {Boolean} isSync  whether request is sync.
+     * @param  {Object}  event   IPC event.
+     * @param  {String}  payload request payload.
+     */
+    _sendRequest (isSync, event, payload) {
+        const ownerId = event.sender.getId();
 
-        this.ipcSocket.on('connect', function(data){
-            _this.sender.send('ipcProvider-setWritable', true);
-            _this.sender.send('ipcProvider-connect', data);
-        });
-
-        // wait for data on the socket
-        this.ipcSocket.on('data', function(data){
-            dechunker(data, function(error, result){
-
-                if (error) {
-                    log.error('IPCSOCKET '+ _this.id +' TIMEOUT ERROR', error);
-                    _this.timeout();
-                    return;
-                }
-
-                // FILTER RESPONSES
-                var event = _this.getResponseEvent(result);
-
-                // if notification, then send it back to the creator of this socket
-                if(!event)
-                    return _this.sender.send('ipcProvider-data', JSON.stringify(result));
-
-                result = _this.filterRequestResponse(result, event);
-
-                // if(result && !_.isArray(result))
-                if(!result.id && !_.isArray(result))
-                    log.debug('IPCSOCKET '+ _this.sender.getId()  +' NOTIFICATION', event.payload, result, "\n\n");
-
-                // SEND SYNC back
-                if(event.sync) {
-                    if(!event.sender.isDestroyed())
-                        event.returnValue = JSON.stringify(result);
-                    delete _this.syncEvents[event.eventId];
-
-                // SEND async back
-                } else {
-                    if(!event.sender.isDestroyed())
-                        event.sender.send('ipcProvider-data', JSON.stringify(result));
-                    delete _this.asyncEvents[event.eventId];
-                }
-            });
-        });
+        log.trace('sendRequest', isSync ? 'sync' : 'async', ownerId, payload);
 
+        let jsonPayload = null;
```

Comment:
are you sure, this is not creating memory leaks?
i always try to avoid using variables outside of functions, like you use this one here in the promise functions..